pyfr/solvers/euler/inters.py

```diff
@@ -124,3 +129,100 @@ def __init__(self, be, lhs, elemap, cfgsect, cfg, bccomm):
 
 class EulerSlpAdiaWallBCInters(EulerBaseBCInters):
     type = 'slp-adia-wall'
+
+
+class MassFlowBCMixin:
+    def __init__(self, be, lhs, elemap, cfgsect, cfg, bccomm):
+        super().__init__(be, lhs, elemap, cfgsect, cfg, bccomm)
+
+        self.c |= self._exp_opts(
+            ['rho', 'u', 'v', 'w'][:self.ndims + 1], lhs
+        )
+
+        self.target_mfr = self.cfg.getfloat(cfgsect, 'mass-flow-rate')
+        self.tstart = self.cfg.getfloat(cfgsect, 'tstart')
+        self.p = self.cfg.getfloat(cfgsect, 'p')
+        self.bcname = cfgsect.removeprefix('soln-bcs-')
+        self.alpha = self.cfg.getfloat(cfgsect, 'alpha')
+        self.eta = self.cfg.getfloat(cfgsect, 'eta')
+        self.nsteps = self.cfg.getint(cfgsect, 'nsteps', 100)
+        self.nflush = self.cfg.getint(cfgsect, 'nflush', 10)
+
+        self._set_external('var_p', 'scalar fpdtype_t')
+        self.tprev = None
+        self.nstep_counter = 0
+        self.nflush_counter = 0
+        self.mf_avg = 0.0
+        
+        surf_list = [(etype, fidx, eidxs) for etype, eidxs, fidx in lhs]
+        self.mf_int = SurfaceIntegrator(cfg, cfgsect, elemap, surf_list)
+
+        if self.bccomm.rank == 0:
+            self.outf = init_csv(self.cfg, self.cfgsect, 't,mf,pbc')
+
+    def calculate_mass_flow(self, solns):
+        ndims, nvars = self.ndims, self.nvars
+        mf = 0.0
+
+        # Get the sizes for the area calculation
+        for etype, fidx in self.mf_int.m0:
+            # Get the interpolation operator
+            m0 = self.mf_int.m0[etype, fidx]
+            nfpts, nupts = m0.shape
+
+            # Extract the relevant elements from the solution
+            uupts = solns[etype][..., self.mf_int.eidxs[etype, fidx]]
+            # Remove unused variables
+            uupts = uupts[:,1:-1,:]
```

Comment:
Can't we just `.sum(axis=...)` here? No need for final :

Reply:
I'm not sure I follow about `.sum(axis=...)`

Reply:
Based on my reading of the code you can sum the three momentum components there-and-then (pre interpolation, which is linear, followed by quadrature, which is linear).